### PR TITLE
Fix static network rendering for flatcar

### LINKED ISSUE #1468: [vSphere] Impossible to deploy machines without static network
With #1263 by @PratikDeoghare the method to check if a provider spec configures static networking was changed from checking if `.Network` is `nil` to a new `IsStaticIPConfig` method. This was necessary because the Spec now includes the `ipFamily` field which is set regardless of whether the network configuration is static or not.

The replacement was done for all userdata providers except `flatcar` which still has the old check in it's config templates
https://github.com/kubermatic/machine-controller/blob/debfca7b52db265625167e1285c9fa7879c4f2d6/pkg/userdata/flatcar/provider.go#L173

Now, when new Machine Deployments are created with the new `ipFamily` setting set (which KKP does by default), the resulting machines all receive a static (but empty) network configuration. This effectively renders them unusable.

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
Fixes a bug where static networking configuration was being wrongfully set for flatcar.

**Which issue(s) this PR fixes**:

Fixes #1468

**What type of PR is this?**

/kind bug

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/userdata/flatcar/provider.go [hunk 1]
```diff
@@ -170,7 +172,7 @@ const userDataIgnitionTemplate = `passwd:
         {{end}}
 {{- end }}
 
-{{- if .ProviderSpec.Network }}
+{{- if .StaticIPConfigProvided }}
```
Comment (PratikDeoghare) [inline pkg/userdata/flatcar/provider.go:175]: You could just `.ProviderSpec.Network.IsStaticIPConfig`

like in https://stackoverflow.com/a/10200273

Comment (ahmedwaleedmalik) [reply pkg/userdata/flatcar/provider.go]: Forgot about this completely 😁

**Review by PratikDeoghare — APPROVED** [review]

**Pull request merged**